lib/dsl/validator.js

```diff
@@ -168,7 +158,9 @@ const configPropsValidations = {
   },
   SKIP_USER_MANAGEMENT: { type: 'BOOLEAN' },
   SKIP_CLIENT: { type: 'BOOLEAN' },
-  SKIP_SERVER: { type: 'BOOLEAN' }
+  SKIP_SERVER: { type: 'BOOLEAN' },
+  UAA_BASE_NAME: { type: 'STRING' },
+  JHIPSTER_VERSION: { type: 'INTEGER' }
```

Comment:
Not sure, it could be `5.0.0`, which is a string. Or `5.0.0-beta3`

Reply:
its definitely string

Reply:
Changed it to string. Note that the pegjs version only supported x.y.z syntax:
```
5.0.0
5.2.4
```

But now there is also support for the suffix part of the semantic version.
At the cost of wrapping this in an actual string literal with its delimiters.
```
"5.0.0"
"5.2.4"
"6.0.0-beta"
```

Reply:
Yeah, this was a shortcoming of mine not thinking about a possible "beta", or "alpha", etc.
About the quotes, if we add them here we must add them everywhere, and I'm against that. Can we do without the quotes?

Reply:
I don't see the quotes element in your PR, am I missing something?

Reply:
> I don't see the quotes element in your PR, am I missing something?

The application config is implemented in a generic manner.
So from a grammar perspective every config key could have any potential value.
https://github.com/jhipster/jhipster-core/blob/master/lib/dsl/jdl_parser.js#L412-L430

The validator.js file adds constraints **post** parsing to create syntax errors if and when
wrong types of values have been provided for specific keys.

This makes it very easy to customize the possible configuration keys and values
without an ever increasing grammar file. So in this specific case there was no need to change the grammar, only the constraints on the grammar.

Reply:
>  About the quotes, if we add them here we must add them everywhere, and I'm against that. Can we do without the quotes?

When you say everywhere, where else do you mean?
Somewhere that only accepts a alphaNumeric name for example would not require any quotes...

It may be possible to accept this without quotes, but it would be risky as it could conflict
with other kinds of tokens,
e.g:
- is "5.2" a decimal number or a version number? 
- is "5.2-beta" a semantic version token or an arithmetic expression "5.2 - beta" (beta is a variable)

 and if we do this at the parser level it would allow whitespace between the parts e.g:
```
5 . 0 .  
3     -  beta
```

Reply:
I think that quotes are only needed when a simple NAME/INTEGER token or combinations of will not suffice.
Just like in JavaScript.

```javascript
myObj.foo = 5 // accessing member using identifier
myObj.["5.0.2-beta"] = 5 // accessing member using string literal and square blocks because '5.0.2-beta" is not a valid javascript identifier.
```

Reply:
Well, aren't we the only ones saying that whatever is after `jhipsterVersion` is a string having a finite form like `[0-9]+\.[0-9]+\.[0-9]+(-[a-z]+)?`?
The JDL won't have variables, ever. This is a 100% certainty. And I can't think of a valid use case where having something like `5.2` would be good for the JDL. This is not a programming language.

Reply:
In that case we combine these 5-7 tokens to a single parser rule to designate a version.
It would still allow whitespace between the different rules as our parser is whitespace insensitive.
But that is a minor concern (a formatter can fix that).

I will try to implement this later today or tomorrow.

Reply:
> It would still allow whitespace between the different rules as our parser is whitespace insensitive.

How so?

@bd82 take your time, I'm trying to get annotations in with the new system

Reply:
O.k. so I'm implementing a semVer **literal** for the jHipsterVersion config key.
And it looks like a simple syntax is easy to implement.

But how **complete** do you expect the solution to be?
The the full specs here:
* https://docs.npmjs.com/misc/semver
* Just the [range syntax](https://docs.npmjs.com/misc/semver#range-grammar) is a full mini grammar by itself.
* tilde / caret modifiers
* Partial numbers ("3.2") is also a semantic version.
* 'X' may be used as a wild card.

Implementing a full parser for semVer is way outside the scope....

So do you prefer limiting the syntax of semVer variants allowed to avoid quotes
and thus potentially limiting the **functionality** of that config property.
Or do you prefer limiting the syntax of the JDL (forcing quotes).
and enabling all functionality of semVer versions?

Reply:
For reference here is a full semVer parser
https://github.com/steveklabnik/semver-parser/tree/master/src

It probably does more than we need, but still provides a feeling of scope.